### PR TITLE
doc: add note on which variables are effective for `runCommand`

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
The build log of the following won't output `foo` as one might expect, but the `$PATH` set by stdenv.

```nix
with import <nixpkgs> {}; runCommand "foo" { PATH = "foo"; } "echo $PATH; touch $out"
```

The first commit is a reformatting that does not change contents.

### CODE REVIEW

File: pkgs/build-support/trivial-builders/default.nix [hunk 1]
```diff
@@ -9,31 +9,35 @@ in
...
 
+     By default `runCommand` runs in a stdenv with no compiler environment.
+     `runCommandCC` uses the default stdenv, `pkgs.stdenv`.
 
-   runCommand "name" {envVariable = true;} ''echo hello > $out''
-   runCommandCC "name" {} ''gcc -o myfile myfile.c; cp myfile $out'';
+     Example:
 
+     runCommand "name" {envVariable = true;} ''echo hello > $out''
+     runCommandCC "name" {} ''gcc -o myfile myfile.c; cp myfile $out'';
```
Comment (roberth) [inline pkgs/build-support/trivial-builders/default.nix:23]:
```suggestion
     ```nix
     runCommand "name" {envVariable = true;} ''echo hello > $out'';
     ```

     ```nix
     runCommandCC "name" {} ''gcc -o myfile myfile.c; cp myfile $out'';
     ```
```

File: pkgs/build-support/trivial-builders/default.nix [hunk 2]
```diff
@@ -9,31 +9,35 @@ in
 
 rec {
 
-  /* Run the shell command `buildCommand' to produce a store path named
-   `name'.  The attributes in `env' are added to the environment
-   prior to running the command. By default `runCommand` runs in a
-   stdenv with no compiler environment. `runCommandCC` uses the default
-   stdenv, `pkgs.stdenv`.
+  /* Run the shell command `buildCommand' to produce a store path named `name'.
```
Comment (roberth) [inline pkgs/build-support/trivial-builders/default.nix:12]: 5 spaces of indentation is editor-unfriendly.

```suggestion
  /*
    Run the shell command `buildCommand' to produce a store path named `name'.
```

**Review by roberth — COMMENTED** [review]
I think the formatting is ok - having a single syntax for multi-line function applications. Is it based on written guidelines yet?

File: pkgs/build-support/trivial-builders/default.nix (hunk 1 above)
Comment (roberth) [inline pkgs/build-support/trivial-builders/default.nix:23]:
```suggestion
     ```nix
     runCommand "name" {envVariable = true;} ''echo hello > $out''
     ```

     ```nix
     runCommandCC "name" {} ''gcc -o myfile myfile.c; cp myfile $out''
     ```
```

**Comment by fricklerhandwerk** [comment]
> Is it based on written guidelines yet?

This was just `nixpkgs-format`, but we'll have https://github.com/NixOS/rfcs/pull/166 soon™.

**Pull request merged**